bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/CTabRendering.java

```diff
@@ -621,17 +622,29 @@ void drawSelectedTab(int itemIndex, GC gc, Rectangle bounds) {
 		}
 
 		if (selectedTabHighlightColor != null) {
+			Color oldBackground = gc.getBackground();
 			gc.setBackground(selectedTabHighlightColor);
 			boolean highlightOnTop = drawTabHighlightOnTop;
 			if (onBottom) {
 				highlightOnTop = !highlightOnTop;
 			}
-			int highlightHeight = 2;
-			int verticalOffset = highlightOnTop ? 0 : bounds.height - (highlightHeight - 1);
-			int horizontalOffset = itemIndex == 0 || cornerSize == SQUARE_CORNER ? 0 : 1;
-			int widthAdjustment = cornerSize == SQUARE_CORNER ? 0 : 1;
-			gc.fillRectangle(bounds.x + horizontalOffset, bounds.y + verticalOffset, bounds.width - widthAdjustment,
-					highlightHeight);
+			if (cornerSize == SQUARE_CORNER) {
+				int highlightHeight = 2 + (superimposeKeylineOutline && highlightOnTop ? OUTER_KEYLINE_WIDTH : 0);
+				int verticalOffset = highlightOnTop ? 0 : outlineBoundsForOutline.height - (highlightHeight - 1);
+				int horizontalOffset = itemIndex == 0 || cornerSize == SQUARE_CORNER ? 0 : 1;
+				int widthAdjustment = cornerSize == SQUARE_CORNER ? 0 : 1;
```

Comment:
Since `cornerSize == SQUARE_CORNER` in this branch, these variable can be completely removed, can't they?